var/spack/repos/builtin/packages/libmongoc/package.py

```diff
@@ -61,12 +62,15 @@ def configure_args(self):
             args.append('--enable-ssl=no')
 
         if spec.satisfies('@1.7.0:'):
-            # --with-{snappy,zlib}=system are currently broken and cause
-            # configure to not find the dependencies. We still want to
-            # explicitly disable them when appropriate.
+            # --with-{snappy,zlib}=system are broken for versions < 1.8.1
             if '+snappy' not in spec:
                 args.append('--with-snappy=no')
+            elif spec.satisfies('@1.8.1:'):
+                args.append('--with-snappy=system')
+
             if '+zlib' not in spec:
                 args.append('--with-zlib=no')
+            elif spec.satisfies('@1.8.1:'):
+                args.append('--with-zlib=system')
```

Comment:
Why linking with system zlib? Same above

Reply:
`system` means "whatever pkg-config returns" in this case, that is, the Spack packages. Otherwise, libmongoc might use its bundled versions.

Reply:
Ah, ok